exercises/bank-account/Example.fs

```diff
@@ -1,24 +1,32 @@
 ﻿module BankAccount
 
-type BankAccount = 
-    | Open of float
-    | Closed
+open System
 
-let mkBankAccount() = Closed
+type BankAccount() = 
+    member val Lock = new Object()
+    member val Balance: float option = None with get,set
 
-let openAccount =
-    function
-    | Open x -> Open x
-    | Closed -> Open 0.0
+let mkBankAccount() = BankAccount()
 
-let closeAccount x = Closed
+let openAccount (account: BankAccount) = 
+    lock account.Lock (fun () ->
+        account.Balance <- Some 0.0
+        account
+    )
 
-let getBalance =
-    function
-    | Open x -> Some x
-    | Closed -> None
+let closeAccount (account: BankAccount) = 
+    lock account.Lock (fun () ->
+        account.Balance <- None    
+        account
+    )
 
-let updateBalance change =
-    function
-    | Open x -> Open (x + change)
-    | Closed -> Closed
\ No newline at end of file
+let getBalance (account: BankAccount) = 
+    lock account.Lock (fun () ->
```

Comment:
No need to lock here are reading from a float is atomic:
https://msdn.microsoft.com/en-us/library/aa691278(VS.71).aspx